zhmcclient/_partition_link.py

```diff
@@ -0,0 +1,752 @@
+                part = self.cpc.partitions.resource_object(
+                    part_uri, {'name': part_name, 'nics': []})
+                if name:
+                    if not re.match(name, part_name):
+                        continue
+                if status:
+                    if part.get_property('status') != status:
+                        continue
+                if part_uri not in part_uri_list:
+                    part_uri_list.append(part_uri)
+                    part_list.append(part)
+        return part_list
+
+    def attach_to_partition(
+            self, partition, properties, wait_for_completion=True,
+            operation_timeout=None):
+        """
+        Attach this partition link to a partition, creating corresponding
+        endpoints in the partition.
+
+        This method performs the "Modify Partition Link" HMC operation and
+        by default waits for completion of its asynchronous job.
+
+        HMC/SE version requirements:
+
+        * SE version >= 2.16.0
+        * for technology-specific support, see the API features described in
+          :ref:`Partition Links`
+
+        Authorization requirements:
+
+        * Object-access permission to this partition link.
+
+        Parameters:
+
+          partition (:class:`~zhmcclient.Partition`): The partition to which
+            this partition link is to be attached.
+
+          properties (dict): TBD
+
+          wait_for_completion (bool):
+            Boolean controlling whether this method should wait for completion
+            of the asynchronous job on the HMC.
+
+          operation_timeout (:term:`number`):
+            Timeout in seconds, for waiting for completion of the asynchronous
+            job on the HMC. The special value 0 means that no timeout is set.
+            `None` means that the default async operation timeout of the
+            session is used. If the timeout expires when
+            `wait_for_completion=True`, a :exc:`~zhmcclient.OperationTimeout`
+            is raised.
+
+        Returns:
+
+          * If `wait_for_completion` is `True`, returns `None`.
+
+          * If `wait_for_completion` is `False`, returns a
+            :class:`~zhmcclient.Job` object representing the asynchronously
+            executing job on the HMC.
+            This job does not support cancellation.
+
+        Raises:
+
+          :exc:`~zhmcclient.HTTPError`
+          :exc:`~zhmcclient.ParseError`
+          :exc:`~zhmcclient.AuthError`
+          :exc:`~zhmcclient.ConnectionError`
+          :exc:`~zhmcclient.OperationTimeout`: The timeout expired while
+            waiting for completion of the operation.
+        """
+        body = {}
+        pl_type = self.get_property('type')
+        if pl_type == 'ctc':
+            body['added-partition-uris'] = [partition.uri]
+            body.update(properties)
+            # properties:
+            #   'devices-per-path'
+            #   'added-paths'
```

Comment:
As part of the `attach_to_partition` you should not need to specify `added-paths` and `devices-per-path`. This information needs to be provided once when creating the partition link to represent the physical connectivity. Unless you do changes to the physical connectivity of your system, you don't need to specify them in the Modify PL call again.

It gets complicated if we want to allow to specify device numbers for the newly attached partitions. Then we would have to dig into the `modified-paths`. But for initial support that's probably not necessary as we automatically generate device numbers if not specified.

Reply:
I have moved the property description into the docstring, and have applied this comment.